plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -101,7 +97,7 @@ public void initializeFromMetadata(TableMetadata tableMetadata)
         currentMetadata = tableMetadata;
         currentMetadataLocation = tableMetadata.metadataFileLocation();
         shouldRefresh = false;
-        version = parseVersion(currentMetadataLocation);
+        version = parseVersion(currentMetadataLocation).orElse(-1);
```

Comment:
If you want as follow up it'd be nice to make the version number Optional here in this class as well, rather than the `orElse(-1)`. Can definitely be a separate PR though

Reply:
Yes, that will be good. I will create a new PR for this change.

Reply:
TODO PR @krvikash